iceoryx_posh/test/moduletests/test_roudi_portmanager.cpp

```diff
@@ -543,7 +559,71 @@ TEST_F(PortManager_test, AcquireConditionVariablesDataAfterDestroyingPreviouslyA
     acquireMaxNumberOfConditionVariables(processName);
 }
 
-TEST_F(PortManager_test, AcquireMaxNumberOfNodePorts)
+TEST_F(PortManager_test, AcquiringMaximumNumberOfEventVariablesWorks)
+{
+    std::string process = "BuddyHolly";
+
+    acquireMaxNumberOfEventVariables(process);
+}
+
+TEST_F(PortManager_test, AcquiringOneMoreThanMaximumNumberOfEventVariableFails)
+{
+    std::string process = "BuddyHollysBrille";
+
+    // first acquire all possible event variables
+    acquireMaxNumberOfEventVariables(process);
+
+    // test if overflow errors get hit
+    auto errorHandlerCalled{false};
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&errorHandlerCalled](const iox::Error, const std::function<void()>, const iox::ErrorLevel) {
+            errorHandlerCalled = true;
+        });
+
+    auto eventVariableDataResult = m_portManager->acquireEventVariableData("AnotherBrille");
+    EXPECT_THAT(eventVariableDataResult.has_error(), Eq(true));
+    EXPECT_THAT(errorHandlerCalled, Eq(true));
+    EXPECT_THAT(eventVariableDataResult.get_error(), Eq(PortPoolError::EVENT_VARIABLE_LIST_FULL));
+}
+
+TEST_F(PortManager_test, DeletingEventVariableWorks)
+{
+    std::string process = "BudSpencer";
+
+    // first acquire all possible event variables
+    acquireMaxNumberOfEventVariables(process);
+
+    // delete one and add one eventVariableDataResult should be possible now
+    unsigned int i = 0U;
+    iox::ProcessName_t newProcessName(iox::cxx::TruncateToCapacity, process + std::to_string(i));
+    m_portManager->deletePortsOfProcess(newProcessName);
+
+    auto eventVariableDataResult = m_portManager->acquireEventVariableData(newProcessName);
+    EXPECT_THAT(eventVariableDataResult.has_error(), Eq(false));
+}
+
+TEST_F(PortManager_test, DestroyEventVariableAndAddNewOneSucceeds)
+{
+    iox::ProcessName_t process = "Terence Hill";
+    std::vector<iox::popo::EventVariableData*> eventVariableContainer;
+
+    // first acquire all possible event variables
+    acquireMaxNumberOfEventVariables(
+        process, [&](auto eventVariableData) { eventVariableContainer.push_back(eventVariableData); });
+
+    // set the destroy flag and let the discovery loop take care
+    for (const auto& eventVariable : eventVariableContainer)
+    {
+        eventVariable->m_toBeDestroyed.store(true, std::memory_order_relaxed);
+    }
+    m_portManager->doDiscovery();
+    eventVariableContainer.clear();
```

Comment:
after the merge with master, there is this nice `setDestroyFlagAndClearContainer` method available which could be used to reduce code duplication